src/skeleton/mod.rs

```diff
@@ -0,0 +1,281 @@
+/// Skeleton structs
+/// Owns json::Animation
+
+mod error;
+mod timelines;
+mod animation;
+
+use json;
+use from_json;
+use std::collections::HashMap;
+use std::io::Read;
+use std::f32::consts::PI;
+use serialize::hex::FromHex;
+
+// Reexport skeleton modules
+pub use self::error::*;
+use self::timelines::*;
+use self::animation::*;
+
+const TO_RADIAN: f32 = PI / 180f32;
+
+fn bone_index(name: &str, bones: &[Bone]) -> Result<usize, SkeletonError> {
+    bones.iter().position(|b| b.name == *name).ok_or(SkeletonError::BoneNotFound(name.into()))
+}
+
+fn slot_index(name: &str, slots: &[Slot]) -> Result<usize, SkeletonError> {
+    slots.iter().position(|b| b.name == *name).ok_or(SkeletonError::SlotNotFound(name.into()))
+}
+
+/// Skeleton data converted from json and loaded into memory
+pub struct Skeleton {
+    /// bones for the skeleton, hierarchically ordered
+    pub bones: Vec<Bone>,
+    /// slots
+    pub slots: Vec<Slot>,
+    /// skins : key: skin name, value: slots attachments
+    pub skins: HashMap<String, Skin>,
+    /// all the animations
+    pub animations: HashMap<String, Animation>
+}
+
+impl Skeleton {
+
+    /// Consumes reader (with json data) and returns a skeleton wrapping
+    pub fn from_reader<R: Read>(mut reader: R) -> Result<Skeleton, String> {
```

Comment:
I suggest `Box<Error>` instead of `String` if you don't want to bother or don't want to expose the underlying error.